lti_consumer/tests/unit/plugin/test_views.py

```diff
@@ -0,0 +1,109 @@
+"""
+Tests for LTI 1.3 endpoint views.
+"""
+from mock import patch
+
+from django.http import HttpResponse
+from django.test.testcases import TestCase
+
+
+class TestLti1p3KeysetEndpoint(TestCase):
+    """
+    Test `public_keyset_endpoint` method.
+    """
+    def setUp(self):
+        super(TestLti1p3KeysetEndpoint, self).setUp()
+
+        self.location = 'block-v1:course+test+2020+type@problem+block@test'
+        self.url = '/lti_consumer/v1/public_keysets/{}'.format(self.location)
+
+        # Patch settings calls to LMS method
+        xblock_handler_patcher = patch(
+            'lti_consumer.plugin.views.run_xblock_handler_noauth',
+            return_value=HttpResponse()
+        )
+        self.addCleanup(xblock_handler_patcher.stop)
+        self._mock_xblock_handler = xblock_handler_patcher.start()
+
+    def test_public_keyset_endpoint(self):
+        """
+        Check that the keyset endpoint works properly.
+        """
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
+
+    def test_invalid_usage_key(self):
+        """
+        Check invalid methods yield HTTP code 405.
```

Comment:
the comment says 405 but the check is for 404?

Reply:
Fixed, it should return a 404.